arcgis-ios-sdk-samples/Maps/Browse building floors/BrowseBuildingFloorsViewController.swift

```diff
@@ -0,0 +1,139 @@
+// Copyright 2021 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import UIKit
+import ArcGIS
+
+class BrowseBuildingFloorsViewController: UIViewController {
+    // MARK: Storyboard views
+    
+    /// The map view managed by the view controller.
+    @IBOutlet var mapView: AGSMapView! {
+        didSet {
+            mapView.map = makeMap()
+        }
+    }
+    /// The floor level picker view.
+    @IBOutlet var floorLevelPickerView: UIPickerView!
+    
+    // MARK: Properties
+    
+    /// The floor levels of the floor-aware web map.
+    var floorLevels: [AGSFloorLevel] = []
+    
+    /// The currently selected floor level.
+    var selectedFloorLevel: AGSFloorLevel? {
+        didSet {
+            // Set the selected level to visible and the previous to invisible.
+            oldValue?.isVisible = false
+            selectedFloorLevel?.isVisible = true
+            // Update picker view selection.
+            let row: Int
+            if let selectedFloorLevel = selectedFloorLevel {
+                row = floorLevels.firstIndex(of: selectedFloorLevel)! + 1
+            } else {
+                row = .zero
+            }
+            floorLevelPickerView.selectRow(row, inComponent: 0, animated: true)
+        }
+    }
+    
+    // MARK: Methods
+    
+    func setPickerViewLayout() {
+        floorLevelPickerView.translatesAutoresizingMaskIntoConstraints = false
+        floorLevelPickerView.layer.cornerRadius = 10.0
+        NSLayoutConstraint.activate([
+            floorLevelPickerView.widthAnchor.constraint(equalToConstant: 120.0),
+            view.safeAreaLayoutGuide.trailingAnchor.constraint(equalToSystemSpacingAfter: floorLevelPickerView.trailingAnchor, multiplier: 1),
+            mapView.attributionTopAnchor.constraint(equalToSystemSpacingBelow: floorLevelPickerView.bottomAnchor, multiplier: 1)
+        ])
+    }
+    
+    /// Create a map.
+    func makeMap() -> AGSMap {
+        // A floor-aware web map for floors of Esri Building L in Redlands.
+        let map = AGSMap(item: AGSPortalItem(
+            portal: .arcGISOnline(withLoginRequired: false),
+            itemID: "f133a698536f44c8884ad81f80b6cfc7"
+        ))
+        map.load { [weak self] error in
+            self?.map(map, didLoadWith: error)
+        }
+        return map
+    }
+    
+    func map(_ map: AGSMap, didLoadWith loadError: Error?) {
+        guard loadError == nil else {
+            presentAlert(error: loadError!)
```

Comment:
Having to force unwrap something checked for `nil` in a guard statement is a code smell and can almost always be avoided. Instead, I would suggest moving this check to where this method (`map(_:didLoadWith:)`) is called:

```swift
if let error = error {
    self?.mapDidLoad(map)
} else {
    self?.presentAlert(error: error)
}
```

Similar with the error in `floorManager(_:didLoadWith:)`.